Assets/Core/Weapon System/Weapon.cs

```diff
@@ -0,0 +1,133 @@
+using UnityEngine;
+using Random = UnityEngine.Random;
+
+/// <summary>
+/// Classify different weapon types
+/// </summary>
+public enum EWeaponType
+{
+    Fire,
+    Melee,
+    Deafault
+}
+
+/// <summary>
+/// This interface will implement all the functions weapons will make use of
+/// </summary>
+public interface IWeapon
+{
+    public EWeaponType GetWeaponType();
+
+    /// <summary>
+    /// Returns the amount of uses (bullets left), -1 if this weapon has no Uses functionality 
+    /// </summary>
+    public int UsesLeft();
+
+    public void Pickup(Transform weaponHolder);
+
+    public void Throw(Vector2 forwardVector);
+
+    public void Use();
+}
+
+/// <summary>
+/// Base weapon class, basic functionality
+/// </summary>
+public class Weapon : MonoBehaviour, IWeapon
```

Comment:
You can make this an abstract class so a game object cannot be created from it, avoiding having to return -1 in `UsesLeft()`